qa/tasks/cephfs/test_volumes.py

```diff
@@ -4654,3 +4654,142 @@ def test_subvolume_snapshot_clone_cancel_pending(self):
 
         # verify trash dir is clean
         self._wait_for_trash_empty()
+
+    def test_malicious_metafile_on_legacy_to_v1_upgrade(self):
+        """
+        Validate handcrafted .meta file on legacy subvol root doesn't break the system
+        on legacy subvol upgrade to v1
+        poor man's upgrade test -- theme continues...
+        """
+        subvol1, subvol2 = self._generate_random_subvolume_name(2)
+
+        # emulate a old-fashioned subvolume in the default group
+        createpath1 = os.path.join(".", "volumes", "_nogroup", subvol1)
+        self.mount_a.run_shell(['mkdir', '-p', createpath1], sudo=True)
```

Comment:
2022-07-27T14:37:36.336 INFO:tasks.cephfs_test_runner:======================================================================
2022-07-27T14:37:36.337 INFO:tasks.cephfs_test_runner:ERROR: test_binary_metafile_on_legacy_to_v1_upgrade (tasks.cephfs.test_volumes.TestVolumes)
2022-07-27T14:37:36.337 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2022-07-27T14:37:36.338 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2022-07-27T14:37:36.338 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_8301e10cc09ad311fdf8cf0b16e30c50653fc6d8/qa/tasks/cephfs/test_volumes.py", line 4722, in test_binary_metafile_on_legacy_to_v1_upgrade
2022-07-27T14:37:36.339 INFO:tasks.cephfs_test_runner:    self.mount_a.run_shell(['mkdir', '-p', createpath], sudo=True)
2022-07-27T14:37:36.339 INFO:tasks.cephfs_test_runner:TypeError: run_shell() got an unexpected keyword argument 'sudo'
2022-07-27T14:37:36.340 INFO:tasks.cephfs_test_runner:
2022-07-27T14:37:36.340 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2022-07-27T14:37:36.340 INFO:tasks.cephfs_test_runner:Ran 6 tests in 398.306s

Reply:
Addressed, hopefully it passes this time.